index.bs

```diff
@@ -1206,6 +1204,32 @@ The <dfn attribute for="XRInputSource">gamepad</dfn> attribute is a {{Gamepad}}
 NOTE: {{XRInputSource}}s with a <code>null</code> {{gamepad}} can still fire {{select}}, {{selectstart}}, and {{selectend}} events to report binary inputs.
 </section>
 
+Transient input {#transient-input}
+-----------------------
+Some [=/XR Device=]s may support <dfn>transient input sources</dfn>, where the [=XR input source=] is only meaningful while performing it's [=primary action=]. An example would be mouse, touch, or stylus input against an {{XRSessionMode/inline}} {{XRSession}}, producing an {{XRInputSource}} with a {{targetRayMode}} set to {{screen}}. This type of [=XR input source=] would only be present in the session's [=list of active XR input sources=] for the duration of the the {{selectstart}}, {{select}}, and {{selectend}} event sequence.
+
+<div class="algorithm" data-algorithm="on-transient-input-start">
+
+When a [=transient input source=] for {{XRSession}} |session| begins it's [=primary action=] the UA MUST run the following steps:
+
+  1. Fire any <code>"pointerdown"</code> events produced by the [=XR input source=]'s action, if necessary.
+  1. [=add input source|Add the XR input source=] to the [=list of active XR input sources=].
+  1. [=Queue a task=] that fires a {{XRInputSourceEvent}} named {{selectstart!!event}} on |session|.
+
+</div>
+
+<div class="algorithm" data-algorithm="on-transient-input-end">
+
+When a [=transient input source=] for {{XRSession}} |session| ends it's [=primary action=] the UA MUST run the following steps:
+
+  1. [=Queue a task=] that fires a {{XRInputSourceEvent}} named {{select!!event}} on |session|.
+  1. Fire any <code>"click"</code> events produced by the [=XR input source=]'s action, if necessary.
+  1. [=Queue a task=] that fires a {{XRInputSourceEvent}} named {{selectend!!event}} on |session|.
```

Comment:
Are we explicitly stating this event ordering for non-transient input sources anywhere in the spec?